src/v3/src/transformer/uischema/createTextElementKeys.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {
+  DescriptionElement,
+  HeadingElement,
+  TitleElement,
+  TransformStepFn,
+} from '../../types';
+import { traverseLayout } from '../util';
+
+const textElementTypes = new Set([
+  'Title',
+  'Heading',
+  'Description',
+]);
+
+export const createTextElementKeys: TransformStepFn = (formbag) => {
```

Comment:
I do this only on text type elements because if more than one is present on the page at once, using the text content apart of the key ensures its uniqueness.

Unless someone knows of a better way to do this without some external package.

Reply:
The only other way I can think of is to generate pretty much entirely random ids as keys

Reply:
I was considering it, but I thought Lester mentioned to avoid using random string/ids for the key. I know doing this on Fields can be problematic (because the components are unmounted and remounted when the key changes) but not so sure how problematic it would be with components that just render text (maybe we still want to avoid an unnecessary re-render?)

Reply:
Yeah I think this approach is fine, I was just spitballing the only other way I could think of (not necessarily that it's a better approach)